src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/Worker.java

```diff
@@ -1134,7 +1138,12 @@ protected ShardConsumer buildConsumer(ShardInfo shardInfo, IRecordProcessorFacto
                         streamConfig.shouldValidateSequenceNumberBeforeCheckpointing()),
                 metricsFactory);
 
-        return new ShardConsumer(shardInfo,
+        if(shardConsumerFactory == null){
+
+            shardConsumerFactory = new KinesisShardConsumerFactory();
+        }
+
+        return shardConsumerFactory.createShardConsumer(shardInfo,
```

Comment:
Will there be any conflict if only shardConsumerFactory is set to Kinesis variant and all other are of different type?

Reply:
I am unsure what you mean by this. The chosen ShardConsumerFactory should create all ShardConsumers so they should all be the same type. Do you mean if the ShardConsumerFactory is the KinesisVariant but other classes like the PeriodicShardSyncManager are the DDB Streams Variant?

Reply:
> Do you mean if the ShardConsumerFactory is the KinesisVariant but other classes like the PeriodicShardSyncManager are the DDB Streams Variant?

yes, will it cause any issue?

Reply:
So a normal KCL application shouldn't have access to anything outside of the standard KCL classes so the shardConsumerFactory and all other classes should be the Kinesis variant. Obviously in an adapter application, using the KinesisShardConsumerFactory with the DDB Streams classes will cause issues but standard KCL applications should never face that issue